infra/Pos/Discovery/Holders.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE DataKinds #-}
+-- | Transformer that carries peer discovery capabilities.
+
+module Pos.Discovery.Holders
+       ( DiscoveryTag
+       , DiscoveryConstT
+       , runDiscoveryConstT
+       , DiscoveryKademliaT
+       , askDHTInstance
+       , runDiscoveryKademliaT
+       ) where
+
+import qualified Control.Monad.Ether              as Ether
+import qualified Control.Monad.Ether              as Ether.E
+import qualified Data.Set                         as S (fromList)
+import           Mockable                         (Async, Catch, Mockables, Promise,
+                                                   Throw)
+import           System.Wlog                      (WithLogger)
+import           Universum
+
+import           Pos.Communication.Types.Protocol (NodeId)
+import           Pos.DHT.Model                    (dhtNodeToNodeId, randomDHTKey)
+import           Pos.DHT.Real                     (KademliaDHTInstance,
+                                                   kademliaGetKnownPeers, kdiHandle,
+                                                   lookupNode)
+import           Pos.Discovery.Class              (MonadDiscovery (..))
+
+----------------------------------------------------------------------------
+-- Common
+----------------------------------------------------------------------------
+
+data DiscoveryTag -- loneliness is something we all know
+
+----------------------------------------------------------------------------
+-- Constant peers
+----------------------------------------------------------------------------
+
+type DiscoveryConstT m = Ether.E.ReaderT DiscoveryTag (Set NodeId) m
+
+instance (Monad m) => MonadDiscovery (DiscoveryConstT m) where
+    getPeers = Ether.E.ask (Proxy @DiscoveryTag)
+    findPeers = pure mempty
```

Comment:
Is it correct? `findPeers` for `DiscoveryKademliaT` returns result of `getPeers`, but this one returns `mempty`. That's why I am asking for documentation.